tachyon/crypto/commitments/fri/two_adic_fri_verifier.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_VERIFIER_H_
+#define TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_VERIFIER_H_
+
+#include <tuple>
+#include <vector>
+
+#include "tachyon/base/bits.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_config.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_proof.h"
+#include "tachyon/math/geometry/dimensions.h"
+#include "tachyon/zk/air/plonky3/challenger/challenger.h"
+
+namespace tachyon::crypto {
+
+template <typename F, typename MMCS>
+struct CommitStep {
+  using Commitment = typename MMCS::Commitment;
+  F beta;
+  Commitment commit;
+  CommitPhaseProofStep<F, MMCS> opening;
+};
+
+template <typename F, typename MMCS>
+F VerifyQuery(size_t index, size_t log_max_height,
```

Comment:
```suggestion
F VerifyQuery(uint32_t index, uint32_t log_max_height,
```

Reply:
I use `row` instead of `height` in `field_merkle_tree_mmcs.h` and If i remembered correctly, it was your suggestion! :)

Reply:
Haha yea I'll change that.